### ISSUE TITLE
GCC 7.1 compile error.

### ISSUE THREAD

**Issue by behdad** [harfbuzz/harfbuzz#3332]
### Discussed in https://github.com/harfbuzz/harfbuzz/discussions/3331

<div type='discussions-op-text'>

<sup>Originally posted by **KP71156** December 10, 2021</sup>
I'm trying to compile HarfBuzz on Linux GCC 7.1 and getting an error in hb-algs.hh:
Harfbuzz/src/hb-algs.hh:336:23: error: expected identifier before ‘&&’ token
   auto operator () (_T&& _v) const HB_AUTO_RETURN \
Code is compiling without problem on Windows.

Pulled latest code from 3.1.2 tag.
Any help would be greatly appreciated.
Ken</div>

**Comment by KP71156** [comment]
Here's the result of g++ -E preprocessor output
[hb-common.cc_prepro.txt](https://github.com/harfbuzz/harfbuzz/files/7694570/hb-common.cc_prepro.txt)
.

**Comment by behdad** [comment]
> Here's the result of g++ -E preprocessor output [hb-common.cc_prepro.txt](https://github.com/harfbuzz/harfbuzz/files/7694570/hb-common.cc_prepro.txt) .

That compiles fine with g++ 7.1 on godbolt... Humm...

Can you try compiling HarfBuzz standalone with this compiler, outside of the bigger project you are building?

**Comment by KP71156** [comment]
Will do.

**Comment by KP71156** [comment]
You are the man! It built fine outside my project, so I pulled out a reference I had added to your CMaketLists.txt file to our own custom CMake settings and now it's building in my project as well. Not sure what setting could have caused this, but now I have an idea how to figure it out. Thanks very much for your help! It did not occur to me that a CMake setting could cause a #define expansion problem.
